common/src/main/java/software/amazon/kms/common/TagHelper.java

```diff
@@ -0,0 +1,178 @@
+package software.amazon.kms.common;
+
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+import com.google.common.collect.Sets;
+import org.apache.commons.collections.MapUtils;
+import org.apache.commons.lang3.ObjectUtils;
+
+import software.amazon.awssdk.services.kms.KmsClient;
+import software.amazon.awssdk.services.kms.model.Tag;
+import software.amazon.cloudformation.proxy.AmazonWebServicesClientProxy;
+import software.amazon.cloudformation.proxy.ProgressEvent;
+import software.amazon.cloudformation.proxy.ProxyClient;
+import software.amazon.cloudformation.proxy.ResourceHandlerRequest;
+
+public class TagHelper<M, C extends KeyCallbackContext, T extends KeyTranslator<M>> {
+    final T keyTranslator;
+    final KeyApiHelper keyApiHelper;
+    final KeyHandlerHelper<M, C, T> keyHandlerHelper;
+
+    public TagHelper(final T keyTranslator, final KeyApiHelper keyApiHelper,
+            final KeyHandlerHelper<M, C, T> keyHandlerHelper) {
+        this.keyTranslator = keyTranslator;
+        this.keyApiHelper = keyApiHelper;
+        this.keyHandlerHelper = keyHandlerHelper;
+    }
+
+    /**
+     * convertToSet
+     *
+     * Converts a tag map to a set of Tag objects.
+     * Note: Like convertToMap, convertToSet filters out value-less tag entries.
+     *
+     */
+    public static Set<Tag> convertToSet(final Map<String, String> tagMap) {
+        if (MapUtils.isEmpty(tagMap)) {
+            return Collections.emptySet();
+        }
+        return tagMap.entrySet().stream()
+                .filter(tag -> tag.getValue() != null)
+                .map(tag -> Tag.builder()
+                        .tagKey(tag.getKey())
+                        .tagValue(tag.getValue())
+                        .build())
+                .collect(Collectors.toSet());
+    }
+
+    /**
+     * generateTagsForCreate
+     *
+     * Generate tags to put into resource creation request.
+     */
+    public final Map<String, String>  generateTagsForCreate(final ResourceHandlerRequest<M> handlerRequest) {
+        final Map<String, String> tagMap = new HashMap<>();
+
+        // merge system tags with desired resource tags if your service supports CloudFormation system tags
+        // KMS does not support system tags currently.
+        // tagMap.putAll(handlerRequest.getSystemTags());
```

Comment:
can we remove this for now and add it later when we actually support it ?

Reply:
Yes, will remove it. These comments got copy-pasted from a resource agnostic sample.